easybuild/tools/toolchain/toolchain.py

```diff
@@ -95,6 +99,10 @@ def __init__(self, name=None, version=None, mns=None):
 
         self.vars = None
 
+        self.add_class_constants_to_restore([])  # make sure self.CLASS_CONSTANTS_TO_RESTORE is initialised
+        self._copy_class_constants()
+        self._restore_class_constants()
```

Comment:
make a new method 
```
def _init_class_constants(self):
  self.add_class_constants_to_restore([])
  self._copy_class_constants()  
  self._restore_class_constants()
``` 

because this looks so strange otherwise